flybywiresim/aircraft
# feat(a380x/pfd): Add FCU ALT ABOVE/BELOW A/C FMA messages

Fixes #9410 

## Summary of Changes

Adds the "FCU ALT ABOVE A/C" and "FCU ALT BELOW A/C" FMA messages

## Screenshots (if necessary)


![Screenshot 2024-12-22 112216](https://github.com/user-attachments/assets/5085b321-0488-4683-a08c-1526d3814663)
![Screenshot 2024-12-22 112341](https://github.com/user-attachments/assets/18db197c-5673-4d14-ac88-23749a925d63)

## References

![Screenshot 2024-12-22 111442](https://github.com/user-attachments/assets/1f1b8299-5ef3-43e9-aaa0-2db88465f0b0)

Issue linked




## Additional context



Discord username (if different from GitHub): Pruznak

## Testing instructions


1. Load up aircraft, take off, climb to any altitude
2. Select the HDG / VS mode on the FCU
3. Select any positive VS, any altitude below current one and engage VS - make sure "FCU ALT BELOW A/C" is displayed on the FMA and does not disappear until you either: 1) Set VS to 0 or 2) Set VS to any negative value or 3) Disengage VS (switch to any other vertical mode)
4. Select any negative VS, any altitude above current one and engage VS - make sure "FCU ALT ABOVE A/C" is displayed on the FMA and disappears only after: 1) roughly 10 seconds have passed since it displayed or 2) You set VS to 0 or 3) You set VS to any positive value or 4) You disengage VS (switch to any other vertical mode)
5. Switch to TRK / FPA mode on the FCU and repeat steps 3 and 4 with FPA
6. Make sure when switching HDG / VS and TRK / FPA modes while either message is displayed doesn't result in it disappearing


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
